src/lib/core/CHIPTLV.h

```diff
@@ -52,6 +52,27 @@
 namespace chip {
 namespace TLV {
 
+struct uint24_t
+{
+    uint24_t() { memset(value, 0, 3); };
+    uint24_t(const uint64_t & v) { memcpy(value, &v, 3); };
```

Comment:
So this is taking the first three bytes of "v"'s memory representation and putting them into our buffer.  On a little-endian system that would be the low three bytes.  On a big-endian system that would be the high three bytes.  That seems suspect.

Reply:
I agree, this is a nasty `hack` only working on some platforms. This would need to be platform agnostic :-)